python/uds.py

```diff
@@ -587,12 +596,13 @@ def _uds_request(self, service_type: SERVICE_TYPE, subfunction: int = None, data
 
     # send request, wait for response
     max_len = 8 if self.sub_addr is None else 7
-    isotp_msg = IsoTpMessage(self._can_client, timeout=self.timeout, debug=self.debug, max_len=max_len)
+    isotp_msg = IsoTpMessage(self._can_client, timeout=self.timeout, debug=self.debug, max_len=max_len, single_frame_mode=True)
```

Comment:
we drain from the can client, so not sure if we can easily test without single frame mode